### DISCUSSION TITLE
limit_type和limit_side_len怎么设定比较好？limit_type=min和limit_side_len=736时，图像最小边小于736时，图像分辨率2480x1396, 有时同行文字会被分割，

### DISCUSSION THREAD

**Discussion by nissansz** [PaddlePaddle/PaddleOCR#14271] [Q&A]
limit_type和limit_side_len怎么设定比较好？limit_type=min和limit_side_len=736时，图像最小边小于736时，图像分辨率2480x1396, 有时同行文字会被分割，
![image](https://github.com/user-attachments/assets/59fdb892-f775-493a-a100-80bbf527d51f)

limit_type=max和limit_side_len=2000时，有时小文字可能检测不到
![image](https://github.com/user-attachments/assets/9d026fdf-956d-4b2a-8364-09074331e9d9)

**Comment by GreatV** [comment]
在使用 `limit_type` 和 `limit_side_len` 参数时，如何优化设置以减少文字分割问题或提高小文字的检测能力，可以参考以下分析和建议：

---

### **1. 问题分析**

1. **`limit_type=min` 和 `limit_side_len=736` 情况：**
   - 当图像的最小边小于 736 时，图像会被缩放处理。
   - 由于缩放后，图像整体分辨率降低（如提到的 2480x1396），文字可能会被压缩得过小，尤其是同行文字之间的间距变小，容易导致检测阶段将文字错误地分割成多段。

2. **`limit_type=max` 和 `limit_side_len=2000` 情况：**
   - 当图像的最大边被限制为 2000 时，虽然整体分辨率较高，但对于小文字或低对比度的文字，检测模型可能无法有效检测到，导致漏检。

3. **切片操作问题（结合相关案例）：**
   - 在高分辨率图片中（如 6624x4662 的工程图），切片操作可以缓解检测中的内存或显存瓶颈，但如果切片的步长（`horizontal_stride`、`vertical_stride`）过大，或者切片重叠区域过小（`merge_x_thres`、`merge_y_thres`），会导致某些文字被分割或遗漏。

---

### **2. 参数优化建议**

#### **(1) 针对 `limit_type` 和 `limit_side_len` 的优化**
- **目标：** 平衡图像分辨率与模型检测能力。
- **建议调整：**
  - 如果文字分割问题明显：
    - 使用 `limit_type="min"`，但将 `limit_side_len` 增加到 **960 或 1024**，以提升图像分辨率，减少同行文字被误分割的情况。
  - 如果小文字漏检：
    - 使用 `limit_type="max"`，但将 `limit_side_len` 降低到 **1500~1800**，以避免图像过度缩放导致小文字检测不到。

#### **(2) 后处理参数优化**
- **目标：** 优化文本框的生成和合并，减少分割和漏检。
- **建议调整：**
  - `det_db_unclip_ratio`：
    - 当前值为 `1.6`，可以适当增大（如 `2.0` 或更高），以扩大检测框，减少同行文字分割的情况。
  - `det_db_box_thresh` 和 `det_db_thresh`：
    - 当前值分别为 `0.5` 和 `0.3`，如果漏检问题明显，可以降低 `det_db_box_thresh`（如 `0.4`），以保留更多的检测框。
  - `det_db_use_dilate`：
    - 如果文字间距较小，开启膨胀操作（`det_db_use_dilate=1`）可能会有改善。

#### **(3) 切片参数优化**
- **目标：** 提高切片的覆盖率，减少漏检。
- **建议调整：**
  - `horizontal_stride` 和 `vertical_stride`：
    - 当前步长为图像宽高的 1/10，可以尝试减小步长（如 1/15 或更小），增加切片的重叠区域。
  - `merge_x_thres` 和 `merge_y_thres`：
    - 当前阈值分别为 20 和 10，若发现切片间的文字未被正确合并，可以适当增大（如 `merge_x_thres=30`，`merge_y_thres=15`）。

---

### **3. 其他优化措施**

1. **增加检测模型的分辨率支持**
   - 在模型配置中，调整参数 `det_limit_side_len`，使其适配更高分辨率的图片（如 1500~2000），以减少压缩导致的文字丢失。

2. **使用增强的数据预处理**
   - 尝试对输入图像进行轻度的图像增强（如对比度增强、锐化），以提升小文字的可见性。

3. **检测与识别的分离优化**
   - 如果文字分割问题依然明显，可以在检测结果中手动调整或合并部分框。

---

### **4. 参考资料**
- **PaddleOCR 参数调优文档：** [PaddleOCR 参数指南](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.1/doc/doc_ch/inference.md)
- **相关讨论：**
  - [limit_type 和 limit_side_len 调整讨论](https://github.com/PaddlePaddle/PaddleOCR/discussions/14271)
  - [图片文本识别漏检问题分析](https://github.com/PaddlePaddle/PaddleOCR/discussions/14241)

---

### **5. 总结推荐**
- **建议初始参数设置：**
  - `limit_type: "min"`
  - `limit_side_len: 960`
  - `det_db_unclip_ratio: 2.0`
  - `det_db_box_thresh: 0.4`
  - `det_db_use_dilate: 1`
  - 切片步长按图像宽高的 `1/15`，增加 `merge_x_thres` 和 `merge_y_thres`。

通过上述优化，应该可以有效减少漏检和同行分割问题，同时提高整体识别效果。

---
*Response generated by 🤖 [feifei-bot](https://github.com/GreatV/feifei) | chatgpt-4o-latest*
